### PR TITLE
Enable testing against NetFX for S.S.C.X509Certificates.Tests

### LINKED ISSUE #30552: S.S.Cryptography.X509Certificates tests do not run against netfx
Apparently we have a couple of deviations.  Since pre-1.0.

### PR DESCRIPTION
**Pull request by bartonjs**
Fixes #30552.

@joperezr I couldn't figure out how to actually get a net462 configuration to build and run the tests.  (Added netfx-Windows_NT to Configurations.props, built `msbuild /t:BuildAndTest /p:TargetGroup=net462`, but that tried launching dotnet.exe when it got to the run tests part)

### CODE REVIEW

File: src/System.Security.Cryptography.X509Certificates/tests/System.Security.Cryptography.X509Certificates.Tests.csproj [hunk 1]
```diff
@@ -4,37 +4,33 @@
   <PropertyGroup>
     <ProjectGuid>{A28B0064-EFB2-4B77-B97C-DECF5DAB074E}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <HaveDSA Condition="'$(TargetGroup)'=='netcoreapp' OR '$(TargetGroup)'=='uap' OR '$(TargetGroup)'=='net462'">true</HaveDSA>
```
Comment (joperezr) [inline src/System.Security.Cryptography.X509Certificates/tests/System.Security.Cryptography.X509Certificates.Tests.csproj]: TargetGroup will never be net462 nor net472. TargetGroup can only be one of the ones in the Configurations.props file of the project. Basically the way it works is that you are building a specific vertical (for example, netfx) and then we will decide which TargetGroup applies best for this specific project, and that will be the value of TargetGroup. For that reason, I would simply just leave all of these conditions as `'$(TargetGroup)' != 'netstandard'`. If you really want to cross compile for net472 or net462, then you have to add those configs into the Configurations.props file

**Comment by joperezr** [comment]
> I couldn't figure out how to actually get a net462 configuration to build and run the tests. 

For that you would need to add a net462 config to Configurations.props, as we basically override the value of TargetGroup to the closest one from configurations.props

**Comment by bartonjs** [comment]
Looks like the "expanded ECC" tests failed, probably a minimum of net47 (I'll have to look to see which version was the boundary condition)

File: src/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs [hunk 2]
```diff
@@ -228,7 +228,16 @@ public static void X509CertificateCollectionThrowsArgumentNullException()
...
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Throws<NullReferenceException>(
+                    () => new X509CertificateCollection.X509CertificateEnumerator(null));
+            }
+            else
+            {
+                Assert.Throws<ArgumentNullException>(
+                    () => new X509CertificateCollection.X509CertificateEnumerator(null));
+            }
```
Comment (stephentoub) [inline src/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs]: There's an AssertExtensions.Throws overload that does this:
https://source.dot.net/#CoreFx.Private.TestUtilities/System/AssertExtensions.cs,109
```C#
AssertExtensions.Throws<ArgumentNullException, NullReferenceException>(() => => new X509CertificateCollection.X509CertificateEnumerator(null));
```

**Comment by bartonjs** [comment]
@dotnet-bot Test Linux arm64 Release Build please (error MSB4166: Child node "3" exited prematurely. Shutting down. ...)

**Pull request merged**